PSReadLine/Keys.cs

```diff
@@ -221,22 +222,36 @@ void AppendPart(string str)
                     if (isShift) { AppendPart("Shift"); }
                     if (isCtrl) { AppendPart("Ctrl"); }
                     if (isAlt) { AppendPart("Alt"); }
-                    AppendPart(key.Key.ToString());
+                    AppendPart(consoleKey.ToString());
                     return sb.ToString();
             }
 
             var c = key.KeyChar;
             var isDeadKey = false;
-            if (char.IsControl(c) )
+            if (char.IsControl(c))
             {
                 // We have the virtual key code and Windows has a handy api to map that to the non-control
                 // character that use for a friendly UI.
                 //
                 // If this fails, we rely on some hard coded control characters below.
                 if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                 {
-                    var keySansControl = new ConsoleKeyInfo(key.KeyChar, key.Key, isShift, isAlt, control: false);
-                    TryGetCharFromConsoleKey(keySansControl, ref c, ref isDeadKey);
+                    // A heuristic to check for dead keys --
+                    //   We got an 'OemXXX' ConsoleKey, '\0' key char, and no 'Ctrl' modifier. It's very likely generated by a dead key.
+                    //   We check for 'Ctrl' modifier because it's easy to generate '\0' KeyChar and 'OemXXX' by combinding 'Ctrl' with
+                    //   another special key, such as 'Ctrl+?' and 'Ctrl+;'.
+                    isDeadKey = (c == '\0') && (consoleKey >= ConsoleKey.Oem1 && consoleKey <= ConsoleKey.Oem102) && !isCtrl;
```

Comment:
There is a large gap of invalid values in that range - maybe you should use a switch?
You also miss `OemClear` in that range.

Reply:
I only see the OemXXX `ConsoleKey`s generated for the dead keys that I have inspected so far ... Yes, `OemClear` is not in that range. I didn't include it only because I didn't see it was generated by the dead keys I have inspected :) I will keep the check as is for now, and change it as needed when there are new dead-key issue reported.